lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/PoolMetadataServer.hs

```diff
@@ -58,13 +58,13 @@ withPoolMetadataServer
     :: (PoolMetadataServer -> ClusterM a)
     -> ClusterM a
 withPoolMetadataServer action = do
-    config@Config{..} <- ask
+    UnliftClusterM withConfig Config{..} <- askUnliftClusterM
     let metadir = untag cfgClusterDir </> "pool-metadata"
     liftIO $ do
         createDirectoryIfMissing False metadir
         withStaticServer metadir $ \baseURL -> do
             let _urlFromPoolIndex i = baseURL </> metadataFileName i
-            runClusterM config $ action PoolMetadataServer
+            withConfig $ action PoolMetadataServer
```

Comment:
Is `ClusterM` worth it compared to taking a `Config` argument? 🤔 Any particular reason for introducing it?

Reply:
Yes , the Config is needed in code "leaves" so passing it around is a bit inconvenient in this case